package test.ipo.task1.service;import java.io.IOException;import org.testng.Assert;import org.testng.annotations.*;import by.ipo.task1.service.FirstFunctionBuilder;public class FirstFunctionBuilderTest {		private FirstFunctionBuilder bff = FirstFunctionBuilder.getInstance();		@DataProvider(name = "buildingData")	public Object[][] setData() {		return new Object[][] { {1, 3, 1, new double[][] {{1, 2, 3},														 {-1, -2, 3}}},								  {-6, 2.4, 2, new double[][] {{-6, -4, -2, 0, 2},									  {6, 4, 2, 0, -2}}},							  };		}		@DataProvider(name = "buildingWrongData")	public Object[][] setWrongData() {		return new Object[][] { {1, 3, 0, new IOException()},								{-9.8, 1.4, -5, new IOException()},							  };		}		@Test(description = "Проверка построения функции 1", 		  dataProvider = "buildingWrongData",		  expectedExceptions = IOException.class)	public void getCoordinatesTestWrong(double lowLimit, double highLimit, 										double step, IOException expectedAnswer) 										   throws IOException {		Assert.assertEquals(bff.getCoordinates(lowLimit, highLimit, step), 							expectedAnswer);	}		@Test(description = "Проверка построения функции 1", 		  dataProvider = "buildingWrongData",		  expectedExceptions = IOException.class)	public void getCoordinatesTestWrong(double lowLimit, double highLimit, 										double step, double[][] expectedAnswer) 											throws IOException {			Assert.assertEquals(bff.getCoordinates(lowLimit, highLimit, step), 								expectedAnswer);		}}